File: app/src/main/java/com/example/myapplication/KalkulatorHargaActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.text.NumberFormat;
import java.util.Locale;

public class KalkulatorHargaActivity extends AppCompatActivity {

    private TextView tvNamaSampah, tvHargaPerUnit, tvTotalHarga;
    private EditText etBerat;
    private Button btnHitung;
    private int hargaSatuan = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_kalkulator_harga);

        tvNamaSampah = findViewById(R.id.tvNamaSampah);
        tvHargaPerUnit = findViewById(R.id.tvHargaPerUnit);
        tvTotalHarga = findViewById(R.id.tvTotalHarga);
        etBerat = findViewById(R.id.etBerat);
        btnHitung = findViewById(R.id.btnHitung);

        // Ambil data dari Intent
        String namaSampah = getIntent().getStringExtra("NAMA_SAMPAH");
        String hargaSampahString = getIntent().getStringExtra("HARGA_SAMPAH"); // e.g., "Rp. 5,000 / Liter"

        // Parse harga dari string (menghilangkan non-digit)
        if (hargaSampahString != null) {
            String hargaAngka = hargaSampahString.replaceAll("[^0-9]", "");
            try {
                hargaSatuan = Integer.parseInt(hargaAngka);
            } catch (NumberFormatException e) {
                hargaSatuan = 0;
            }
        }

        // Tampilkan info
        tvNamaSampah.setText(namaSampah);
        tvHargaPerUnit.setText(hargaSampahString);

        // Listener untuk tombol hitung
        btnHitung.setOnClickListener(v -> hitungTotal());

    }

    private void hitungTotal() {
        String beratString = etBerat.getText().toString();
        if (beratString.isEmpty()) {
            tvTotalHarga.setText("Rp. 0");
            return;
        }

        try {
            double berat = Double.parseDouble(beratString);
            double total = berat * hargaSatuan;

            // Format ke Rupiah
            NumberFormat formatRupiah = NumberFormat.getCurrencyInstance(new Locale("id", "ID"));
            tvTotalHarga.setText(formatRupiah.format(total));

        } catch (NumberFormatException e) {
            tvTotalHarga.setText("Input tidak valid");
        }
    }
}

File: app/src/main/java/com/example/myapplication/KalkulatorHargaActivity.kt
package com.example.myapplication

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.myapplication.databinding.ActivityKalkulatorHargaBinding
import java.text.NumberFormat
import java.util.Locale

class KalkulatorHargaActivity : AppCompatActivity() {

    private lateinit var binding: ActivityKalkulatorHargaBinding
    private var hargaSatuan = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityKalkulatorHargaBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Ambil data dari Intent
        val namaSampah = intent.getStringExtra("NAMA_SAMPAH")
        val hargaSampahString = intent.getStringExtra("HARGA_SAMPAH") // e.g., "Rp 5.000"

        // Parse harga dari string (menghilangkan non-digit)
        hargaSampahString?.let {
            val hargaAngka = it.replace(Regex("[^0-9]"), "")
            hargaSatuan = hargaAngka.toIntOrNull() ?: 0
        }

        // Tampilkan info
        binding.tvNamaSampah.text = namaSampah
        binding.tvHargaPerUnit.text = hargaSampahString

        // Listener untuk tombol hitung
        binding.btnHitung.setOnClickListener { hitungTotal() }
    }

    private fun hitungTotal() {
        val beratString = binding.etBerat.text.toString()
        if (beratString.isEmpty()) {
            binding.tvTotalHarga.text = formatRupiah(0.0)
            return
        }

        val berat = beratString.toDoubleOrNull()
        if (berat == null) {
            binding.tvTotalHarga.text = "Input tidak valid"
            return
        }

        val total = berat * hargaSatuan
        binding.tvTotalHarga.text = formatRupiah(total)
    }

    private fun formatRupiah(nilai: Double): String {
        val formatRupiah = NumberFormat.getCurrencyInstance(Locale("in", "ID"))
        return formatRupiah.format(nilai)
    }
}
